controllers/clustermodule_reconciler_test.go

```diff
@@ -72,6 +73,66 @@ func TestReconciler_Reconcile(t *testing.T) {
 				g.Expect(ctx.VSphereCluster.Spec.ClusterModules).To(gomega.HaveLen(2))
 			},
 		},
+		{
+			name: "when cluster modules already exist but vCenter returns an error",
+			clusterModules: []infrav1.ClusterModule{
+				{
+					ControlPlane:     true,
+					TargetObjectName: "kcp",
+					ModuleUUID:       kcpUUID,
+				},
+				{
+					ControlPlane:     false,
+					TargetObjectName: "md",
+					ModuleUUID:       mdUUID,
+				},
+			},
+			haveError: true,
+			setupMocks: func(svc *cmodfake.CMService) {
+				svc.On("DoesExist", mock.Anything, mock.Anything, kcpUUID).Return(false, vCenter500err)
+				svc.On("DoesExist", mock.Anything, mock.Anything, mdUUID).Return(false, vCenter500err)
+			},
+			customAssert: func(g *gomega.WithT, ctx *context.ClusterContext) {
+				g.Expect(ctx.VSphereCluster.Spec.ClusterModules).To(gomega.HaveLen(2))
+				// Ensure the old modules still exist.
+				g.Expect(ctx.VSphereCluster.Spec.ClusterModules[0].ModuleUUID).To(gomega.BeElementOf(kcpUUID, mdUUID))
+				g.Expect(ctx.VSphereCluster.Spec.ClusterModules[1].ModuleUUID).To(gomega.BeElementOf(kcpUUID, mdUUID))
+				// Check that condition got set.
+				g.Expect(conditions.Has(ctx.VSphereCluster, infrav1.ClusterModulesAvailableCondition)).To(gomega.BeTrue())
+				g.Expect(conditions.IsFalse(ctx.VSphereCluster, infrav1.ClusterModulesAvailableCondition)).To(gomega.BeTrue())
+				g.Expect(conditions.Get(ctx.VSphereCluster, infrav1.ClusterModulesAvailableCondition).Message).To(gomega.ContainSubstring(vCenter500err.Error()))
+			},
+		},
+		{
+			name: "when cluster modules do not exist anymore",
+			clusterModules: []infrav1.ClusterModule{
+				{
+					ControlPlane:     true,
+					TargetObjectName: "kcp",
+					ModuleUUID:       kcpUUID,
+				},
+				{
+					ControlPlane:     false,
+					TargetObjectName: "md",
+					ModuleUUID:       mdUUID,
+				},
+			},
+			setupMocks: func(svc *cmodfake.CMService) {
+				svc.On("DoesExist", mock.Anything, mock.Anything, kcpUUID).Return(false, nil)
+				svc.On("DoesExist", mock.Anything, mock.Anything, mdUUID).Return(false, nil)
```

Comment:
```suggestion
				svc.On("DoesNotExist", mock.Anything, mock.Anything, kcpUUID).Return(false, nil)
				svc.On("DoesNotExist", mock.Anything, mock.Anything, mdUUID).Return(false, nil)
```

Reply:
DoesExist is the name of the function to mock :-) We can't change it here.

Reply:
The mock does even make the test to fail if:
* A function is called which was not mocked with the given input (e.g. `mdUUID` differs)
* A function was mocked but never called

Reply:
Ah :D Sorry I thought this is some sort of description :)